### PR TITLE
Added finnhub python example

### PR DESCRIPTION
**Pull request by pdamodaran**
This example creates a publisher and subscriber using the FinnHub API, Ensign, and River.

### CODE REVIEW

File: finnhub/python/README.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
+# OpenSky Python
```
Comment (daniellemaxwell) [inline finnhub/python/README.md]:
```suggestion
# FinnHub Python
```

File: finnhub/python/README.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+Connect to the Finnhub API to get access to real-time stock market data and play around with generating your own trading and investment strategies!
+
+## Setup
+
+FinnHub requires the `websockets` package for connecting to its API and the example code will use on the `pyensign` package to create a publisher that will collect data from FinnHub and a subscriber that will run a model pipeline using the `river` package to generate predictions.
```
Comment (daniellemaxwell) [inline finnhub/python/README.md]:
```suggestion
FinnHub requires the `websockets` package for connecting to its API and the example code will use the `pyensign` package to create a publisher that will collect data from FinnHub and a subscriber that will run a model pipeline using the `river` package to generate predictions.  
```

File: finnhub/python/README.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+topics:read
+```
+
+Alternatively, specify Ensign credentials in the environment.
+```
+$ export ENSIGN_CLIENT_ID=<your-client-id>
+$ export ENSIGN_CLIENT_SECRET=<your-client-secret>
+```
+
+Get a [free API key](https://finnhub.io/dashboard) from FinnHub. Specify this API key in the environment
```
Comment (daniellemaxwell) [inline finnhub/python/README.md]:
```suggestion
Get a [free API key](https://finnhub.io/dashboard) from FinnHub. Specify this API key in the environment.
```

File: finnhub/python/README.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+```python
+TradesPublisher(symbols=[<your-custom-symbols>])
+```
+
+## Subscribe to Trade Events and Publish predictions to a new topic
+
+Create an [Ensign API key](https://rotational.app) with the permissions:
+
+```json
```
Comment (daniellemaxwell) [inline finnhub/python/README.md]: Should `json` be on this line?

File: finnhub/python/README.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+```
+
+Create a subscriber and start consuming events.
+
+```python
+subscriber = TradesSubscriber()
+subscriber.run()
+```
+
+The subscriber takes the events and runs a model pipeline that generates trade predictions and publishes the predictions to a new `predictions` topic. You will periodically messages being printed to the screen that displays the symbol, timestamp, price, and the predicted price.
```
Comment (daniellemaxwell) [inline finnhub/python/README.md]:
```suggestion
The subscriber takes the events and runs a model pipeline that generates trade predictions and publishes the predictions to a new `predictions` topic. You will periodically see messages being printed to the screen that display the symbol, timestamp, price, and the predicted price.
```

File: finnhub/python/subscriber.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        timestamp = datetime.fromtimestamp(epoch_time)
+        return timestamp
+    
+    async def run_model_pipeline(self, event):
+        """
+        Train an online model and publish predictions to a new topic.
+        Run your super smart model pipeline here!
+        """
+        data = json.loads(event.data)
+        # convert unix epoch to datetime
```
Comment (daniellemaxwell) [inline finnhub/python/subscriber.py:53]: Super useful comments here!

File: finnhub/python/publisher.py [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+class TradesPublisher:
+    """
+    TradesPublisher queries an API for trading updates and publishes events to Ensign.
+    """
+
+    def __init__(self, symbols=["AAPL", "MSFT", "AMZN"], topic="trades"):
+        self.symbols = symbols
+        self.topic = topic
+        self.ensign = Ensign()
```
Comment (daniellemaxwell) [inline finnhub/python/publisher.py:21]: Would the `Ensign credentials` go inside `Ensign` if they're specified in the environment?

**Review by daniellemaxwell — APPROVED** [review]
It was great to see how pyensign works with the Finnhub API! I think anyone looking at this example will really appreciate the comments. I only had some minor suggestions, but this is good to go when you're ready.

File: finnhub/python/publisher.py (hunk 7 above)
Comment (pdamodaran) [reply finnhub/python/publisher.py:21]: if the client id and secret in set in the environment variables, then they don't have to be specified in the call to `Ensign()`.

File: finnhub/python/README.md (hunk 4 above)
Comment (pdamodaran) [reply finnhub/python/README.md]: good catch!  I removed it.

**Comment by pdamodaran** [comment]
Thank you so much for the thorough review and catching all my typos! ;-)

**Pull request merged**